apps/extension/src/core/views/ActivityItem.tsx

```diff
@@ -13,35 +12,37 @@ import type { Transaction } from "~core/managers/transaction"
 import { transactionManager } from "~core/managers/transaction"
 import { formatDate } from "~core/utils/utils"
 
-function createMediaDownloadLinks(transaction: Transaction) {
-  if (!transaction.outputs) {
+function createMediaDownloadLinks({ input, outputs }: Transaction) {
+  if (!outputs) {
     return null
   }
-  const input = transaction.input
-  // fallback to .txt containing contents
-  let mimeType: MediaMimeType
-  const outputs = transaction.outputs
-  if (!isPromptInput(input) || !outputs.every(isMediaOutput)) {
-    return null
+  if(!outputs.every(isMediaHosted)){
+    return "Due to storage constraints, local media generations are currently not downloadable here, but may be soon."
   }
-  //check if a valid mimetype is contained in the data uri
-  const mimeTypeFromUri = outputs[0].uri.split(";")[0].split(":")[1]
-  if (isMediaMimeType(mimeTypeFromUri)) {
-    mimeType = mimeTypeFromUri
+  if (!isPromptInput(input)) {
+    return null
   }
-
   return outputs.map((output, index) => {
-    const fileName = `${input.prompt.replace(" ", "-")}-${index}${mimeType ? "." : ""}${mimeTypeToExtension(mimeType)}`
+    // defaults to no extension
+    let extension: string = ""
+    if (output.url) {
+      const base = output.url.split("?")[0]
```

Comment:
better to use `new URL(output.url)` and then use the `pathname` or whatever directly

Reply:
changed